test/open_telemetry_decorator/attributes_test.exs

```diff
@@ -91,6 +91,10 @@ defmodule OpenTelemetryDecorator.AttributesTest do
     test "handles nested attributes" do
       assert Attributes.get([obj: %{id: 1}], [[:obj, :id]]) == [obj_id: 1]
     end
+    
+    test "handles nested structs" do
+      assert Attributes.get([obj: Date.new(2000, 1, 1)], [[:obj, :year]]) == [obj_year: 2000]
```

Comment:
`Date.new` returns a tuple, and `Date.new!` passes because it's only one level deep. I can reproduce it with a struct that contains a struct though. Github lets me checkout your PR, I'll see if I can update it and fix the formatting.

Thanks for finding this!

```
assert Attributes.get([obj: %SomeStruct{failed: %SomeStruct{count: 3}}], [[:obj, :failed, :count]]) == [obj_failed_count: 3]
```

Reply:
Ahh got it!